docs/better-sandwiches/index.md

```diff
@@ -0,0 +1,290 @@
+  | Sandwich(sandwich) => 10. // [!code ++]
+  | Burger => 15.
+  | Hotdog => 5.;
+
+let toEmoji =
+  fun
+  | Sandwich => {js|🥪|js} // [!code --]
+  | Sandwich(sandwich) => {js|🥪|js} // [!code ++]
+  | Burger => {js|🍔|js}
+  | Hotdog => {js|🌭|js};
+```
+
+Now there's a different error:
+
+```
+File "src/order-confirmation/Item.re", line 13, characters 13-20:
+13 |   | Sandwich(sandwich) => 10.
+                  ^^^^^^^^
+Error (warning 27 [unused-var-strict]): unused variable sandwich.
+```
+
+## A valid use of wildcard
+
+OCaml demands that we use variables we declare[^1], unless they begin with
+underscore. So we can silence the error by renaming `sandwich` to `_sandwich`:
+
+```reason
+| Sandwich(_sandwich) => 10.
+```
+
+However, this essentially turns `_sandwich` into a wildcard, which is [generally
+a bad idea](/order-confirmation/#wildcard-in-switch-expressions). The reality,
+though, is that the warning against using wildcards is more a guideline than a
+hard and fast rule. In this case, it's useful to temporarily use wildcards to
+silence the compiler errors while we fill in the missing logic for
+`Item.toPrice` and `Item.toEmoji`. Still, when the wildcard is meant to be
+temporary, we can give it a more attention-grabbing name so it's less likely to
+become permanent:
+
+```reason
+| Sandwich(_todo_please_use_this_sandwich) => 10.
+```
+
+## Update `Item.toPrice` function
+
+The price of a sandwich should vary depending on the type of sandwich received
+by `Item.toPrice`[^2]:
+
+<<< Item.re#to-price-nested-switch
+
+Actually, there's no need for the nested switch, as OCaml allows you to pattern
+match into nested structures:
+
+<<< Item.re#to-price-flat
+
+## Combining branches in switch
+
+Madame Jellobutter decides to upgrade her portabellos from canned mushrooms to
+artisanal mushrooms lovingly grown by mustachioed hipsters on a local farm,
+which increases the price of portabello sandwiches to $10. Now portabello and
+ham sandwiches are the same price, so you update the `Item.toPrice` function
+accordingly:
+
+<<< Item.re#to-price-same-price{3}
+
+We can combine the portabello and ham branches because they return the same
+value:
+
+<<< Item.re#to-price-combine{3-4}
+
+Since the outer constructor `Sandwich` is the same in both branches, we can
+combine them into a single branch:
+
+<<< Item.re#to-price{3}
+
+The power of pattern matching gives us both clarity and brevity. This is the
+OCaml Way!
+
+## Update `Item.toEmoji` function
+
+`Item.toEmoji` should now return a combination of two emojis when it gets a
+`Sandwich(_)` argument:
+
+| Sandwich | Emoji |
+| -------- | ----- |
+| Portabello | 🥪(🍄) |
+| Ham | 🥪(🐷) |
+| Unicorn | 🥪(🦄) |
+
+We can implement this using `Js.Array.joinWith`:
+
+<<< Item.re#to-emoji-joinwith{3-13}
+
+This touches on the issue of coding style for switch expressions. When possible,
+put shorter branches before longer branches:
+
+<<< Item.re#to-emoji-short-branch-first
+
+## `j` quoted string literal
```

Comment:
I know the full term is something like "quoted string literal using the quoted string identifier j" but that seemed too long, so in most places, I shortened it to "j quoted string literal".

Reply:
Yeah I think what you suggested is more clear, especially for those not familiar with what "quoted strings" is, which will prob be the case.

Reply:
Changed "j quoted string literal" to "{j||j} quoted string literal" in https://github.com/melange-re/melange-for-react-devs/pull/13/commits/0fd698d595fa7a9eb552ae6a160a43cf4345ee0f